src/lang-promql/parser/parser.ts

```diff
@@ -83,33 +83,28 @@ export class Parser {
   analyze() {
     // when you are at the root of the tree, the first node is not `Expr` but a node with no name.
     // So to be able to iterate other the node relative to the promql node, we have to get the first child at the beginning
-    this.checkAST(this.tree.firstChild);
+    this.checkAST(this.tree.topNode.firstChild);
     this.diagnoseAllErrorNodes();
   }
 
   private diagnoseAllErrorNodes() {
-    this.tree.iterate({
-      enter: (type, start, end) => {
-        // usually there is an error node at the end of the expression when user is typing
```

Comment:
Could you keep this comment? Makes it much easier to understand the purpose :)

Reply:
done